pkg/test/framework/components/zipkin/kube.go

```diff
@@ -322,3 +371,17 @@ func buildSpan(obj any) Span {
 	}
 	return s
 }
+
+func resolveHostDomainToIP(hostDomain string) (string, error) {
+	ips, err := net.LookupIP(hostDomain)
+	if err != nil {
+		return "", err
+	}
+	// Use the first resolved IP address
+	for _, ip := range ips {
+		if ipv4 := ip.To4(); ipv4 != nil {
```

Comment:
should we consider IPv6 here?

Reply:
I could add the support for IPv6 is needed.